tests/tests_data/streaming/test_data_processor.py

```diff
@@ -193,6 +195,26 @@ def fn(*_, **__):
     assert os.listdir(cache_dir) == ["a.txt"]
 
 
+def test_wait_for_disk_usage_higher_than_threshold(monkeypatch):
+    class Usage:
+        def __init__(self, free):
+            self.free = free * 1000 * 1000 * 1000
+
+    usages = [Usage(1), Usage(1), Usage(100)]
+
+    def fn(*_, **__):
+        value = usages.pop(0)
+        if value is None:
+            return value
+        return value
+
+    monkeypatch.setattr(shutil, "disk_usage", fn)
```

Comment:
You should patch where you use it, not where it is defined originally. So that means patching

lightning.data.streaming.data_processor.whatever

Also, the whole thing can be done in 2 lines of code like so (sketch):

```py
disk_usage_mock = Mock(side_effect=[Mock(free=1), Mock(free=1), Mock(free=100)])
mock.patch("lightning.data.streaming.data_processor.shutil.disk_usage", disk_usage_mock)
```

and then later you can `assert disk_usage_mock.call_count == 3` or whatever. And probably you want to extend the test a bit, to test the threshold value.